code/modules/projectiles/ammunition.dm

```diff
@@ -82,7 +89,7 @@
 	if(istype(A, /obj/item/ammo_casing)) //loading a bullet into the magazine or box
 		var/obj/item/ammo_casing/AC = A
 		var/accepted = 0
-		if((exact && (AC.type == text2path(ammo_type))) || (!exact && (AC.caliber == caliber)))
+		if((exact && (istype(AC, ammo_type))) || (!exact && (AC.caliber == caliber)))
```

Comment:
Just curious. Why did you do this? It must not have been because something here needed it, because it doesn't work.